### ISSUE TITLE
Broken links in Sparql lesson

### ISSUE THREAD

**Issue by bmschmidt** [programminghistorian/jekyll#289]
There are at least two broken links in the graph-databases-and-SPARQL.md lesson. One is to http://rdf.myexperiment.org/howtosparql? at the end. More critically, the page linked under 'Europeana's' in the following quote leads to a blank page, making it hard to follow along.

---

Up until now, we have constructed queries that look for patterns in one dataset
alone. In the ideal world envisioned by Linked Open Data advocates, multiple
databases can be interlinked to allow very complex queries dependent on
knowledge present in different locations. However, this is easier said than
done, and many endpoints (the BM's included) do not yet reference outside
authorities.

One endpoint that does, however, is
[Europeana's](http://europeana.ontotext.com/sparql). They have created links
between the objects in their database and records about individuals in
[DBPedia](http://wiki.dbpedia.org/) and [VIAF](https://viaf.org/), places in
[GeoNames](http://sws.geonames.org/), and concepts in the Getty Art &
Architecture thesaurus. SPARQL allows you to insert `SERVICE` statements that
instruct the database to "phone a friend" and run a portion of the query on
an outside dataset, using the results to complete the query on the local
dataset. While this lesson will go into the data models in Europeana and DBpedia in depth, the following query illustrates how a `SELECT` statement works. You may run it yourself by copying and pasting the query text into the [Europeana endpoint](http://europeana.ontotext.com/sparql).

**Comment by mdlincoln** [comment]
Thanks for submitting this.
1. I would probably just remove http://rdf.myexperiment.org/howtosparql
2. This is _sigh_ a more serious issue. It looks like Europeana is no longer supporting the SPARQL endpoint, at least not in the form it was in when I wrote the lesson:
   ![screen shot 2016-09-11 at 7 33 46 pm](https://cloud.githubusercontent.com/assets/4392604/18423438/e734cf64-7859-11e6-80c9-0307d0261d0e.png)
   I'm reaching out to Europeana Labs to check on the story behind the endpoint, but it _was_ a "beta" service, and so may truly be down for good. If this is so, I'd suggest adding some type of notice at this point in the lesson that the service has been deprecated. It might even be a good space to point out one of the common weaknesses of "linked open data" services...

Any thoughts, editors?

**Comment by acrymble** [comment]
Thanks for flagging @bmschmidt. Let us know how Europeana responds @mdlincoln. That's the best first approach.

**Comment by mdlincoln** [comment]
So the most detail they were able to share with me at this point is that they are working on a replacement endpoint, but that they do not have a timeline for when it will be back up, nor is there any news on whether it will use the same data mapping as it did when I authored the query used in this lesson.

I'll go ahead and put together a PR switching the one broken link, and adding a notice to the "Linking multiple SPARQL endpoints" section

**Comment by acrymble** [comment]
This is why I publish datasets instead of websites. We've had a similar problem with the Old Bailey Online: #285 . Editors, please keep this in mind as new lessons are being produced. This is not a good use of our time.

**Comment by acrymble** [comment]
@mdlincoln can you email me the contact address you used at Europeana. I'm going to lobby them to see if we can get them to produce a solution. If international libraries cannot offer consistent access to services then who can?

**Comment by acrymble** [comment]
Europeana has a grant call out right now to use their resources. Up too 8k euros. @mdlincoln do you think this is a fit that could fix the problem? Or is it much bigger than that? http://research.europeana.eu/blogpost/europeana-research-grants-2016

**Comment by mdlincoln** [comment]
@acrymble The folks at Europeana Labs did confirm with me that a new version of their SPARQL endpoint is being developed, but that it will be several months from now. It's an internal project, so it wouldn't fit with that grant (also it'd be quite a bit costlier than 8k€!)

**Comment by mdlincoln** [comment]
I was just notified that the Europeana SPARQL endpoint has been reactivated at a new address (blog post here): http://labs.europeana.eu/blog/europeana-sparql-endpoint

In about three weeks once I finish off some travel, I'll be able to check if the same queries from the lesson will work on this new endpoint. If so, I can update the lesson and just make sure it points to their new endpoint host. If they've made major changes to underlying ontologies they're using, though, I may not have the bandwidth to fully re-write the SPARQL queries.
